### ISSUE TITLE
cmd/compile: internal compiler error: large ONEW with EscNone: new()

### ISSUE THREAD

**Issue by xi2** [golang/go#11728]
Go version: +2e4b659 15/07/2015, OS: Debian Jessie AMD64

The following program fails to compile:

``` go
package main

type X struct {
    Array [1 << 16]byte
}

func main() {
    _ = &X{}
}
```

with the message

```
./test.go:8: internal compiler error: large ONEW with EscNone: new()
```

**Comment by dr2chase** [comment]
You can work around that error by causing the pointer to escape.

**Comment by nightlyone** [comment]
Note: This is a regression. It doesn't happen with Go 1.4.2

**Comment by dr2chase** [comment]
It's probably @rsc 's call if we want to try to tweak this for 1.5, but there was a lot of work done for escape analysis and some lurking bugs plugged.  For this particular case (large data with no pointers allocated on stack) it could probably be allowed to pass, but if pointers are in the large data a phase-after-escape-analysis moves such large stack allocations to the heap, potentially leaking stack pointers into the heap.  That is my recollection, at least -- we replaced a potential difficult-to-analyze Heisenbug with a compile-time-error with a workaround.

**Comment by paulmach** [comment]
Changing `_ = &X{}` to `_ = new(X)` allows it to compile. ie. the below works fine.
```
package main

type X struct {
	Array [1 << 16]byte
}

func main() {
	_ = new(X)
}
```

**Comment by ALTree** [comment]
Same as #15733, fixes by d603c27c6b462f044a7079ce5113d90bb3ca4814 since go1.7.
